test/toPaths.spec.js

```diff
@@ -149,5 +151,65 @@ describe('toPaths', function() {
 
         expect(toTree(toPaths(treeMap))).to.deep.equals(expectedTree);
     });
-});
 
+    describe('isSafeNumber', function() {
+
+        var thingsThatShouldReturnTrue = [
+            0,
+            1,
+            -0,
+            -1,
+            10,
+            -10,
+            9007199254740991, // max safe int
+            -9007199254740991, // min safe int
+            '0',
+            '1',
+            '-1',
+            '10',
+            '-10',
+            '9007199254740991', // max safe int
+            '-9007199254740991', // min safe int
+        ];
+        thingsThatShouldReturnTrue.forEach(function(thing) {
+            var should = 'should return true on ' + JSON.stringify(thing);
+            it(should, function() {
+                expect(isSafeNumber(thing)).to.equal(true);
+            });
+        });
+
+        var thingsThatShouldReturnFalse = [
+            [],
+            {},
+            null,
+            true,
+            false,
+            undefined,
+            Infinity,
+            -Infinity,
+            '9007199254740992', // max safe int + 1
+            '-9007199254740992', // min safe int - 1
+            9007199254740992, // max safe int + 1
+            -9007199254740992, // min safe int - 1
+            '648365838265483646384563538',
```

Comment:
@michaelbpaulson - Yup, it has some numbers that are out of the safe range by 1, and now also an absurdly large string number.